src/layout/hierarchy/index.ts

```diff
@@ -97,10 +122,12 @@ export const Layout = () => {
       return edgeIndex
     }, {})
 
-    if (edgeIndex[root] === undefined) {
+    const parent = edgeIndex[root] === undefined ? findParent(graph.nodes, root) : undefined
+    if (edgeIndex[root] === undefined && parent === undefined) {
```

Comment:
w/ the above change, this can be simplified to
```ts
const root = findAncestor(graph.nodes, root)

if (edgeIndex[root] === undefined) {
  return ...
}

...

graph.options?.bfs !== false ?
  graphToBFSHierarchy(edgeIndex, root) :
  graphToDFSHierarchy(edgeIndex, root)
```